src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Queries/GetProviderPaymentsPriority/GetProviderPaymentsPriorityQueryHandler.cs

```diff
@@ -0,0 +1,110 @@
+﻿using FluentValidation;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.CommitmentsV2.Data;
+using SFA.DAS.CommitmentsV2.Data.Expressions;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.CommitmentsV2.Application.Queries.GetProviderPaymentsPriority
+{
+    public class GetProviderPaymentsPriorityQueryHandler : IRequestHandler<GetProviderPaymentsPriorityQuery, GetProviderPaymentsPriorityQueryResult>
+    {
+        private readonly Lazy<ProviderCommitmentsDbContext> _dbContext;
+        private readonly ILogger<GetProviderPaymentsPriorityQueryHandler> _logger;
+
+        public GetProviderPaymentsPriorityQueryHandler(Lazy<ProviderCommitmentsDbContext> dbContext,
+            ILogger<GetProviderPaymentsPriorityQueryHandler> logger)
+        {
+            if (dbContext == null)
+                throw new ArgumentNullException(nameof(dbContext));
+            if (logger == null)
+                throw new ArgumentNullException(nameof(logger));
+
+            _dbContext = dbContext;
+            _logger = logger;
+        }
+
+        public async Task<GetProviderPaymentsPriorityQueryResult> Handle(GetProviderPaymentsPriorityQuery message, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"Getting Provider Payment Priority for employer account {message.EmployerAccountId}");
+
+            // get the approved providers and their current custom provider payment priority order (if any)
+            // ordered by their approved on date for the given employer account
+            var query = from customProviderPaymentPriority in 
```

Comment:
The sql for this is quite complex hence this linq is complex as well, can you please raise tech-debt task for this to see if this can be simplified.